apps/rush-lib/src/plugins/AzureStorageBuildCachePlugin/index.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { Import, JsonSchema } from '@rushstack/node-core-library';
+import { IRushPlugin } from '../../pluginFramework/IRushPlugin';
+import { RushSession } from '../../pluginFramework/RushSession';
+import { RushConfiguration } from '../../api/RushConfiguration';
+
+import type { AzureEnvironmentNames } from './AzureStorageBuildCacheProvider';
+
+const AzureStorageBuildCacheProviderModule: typeof import('./AzureStorageBuildCacheProvider') = Import.lazy(
+  '../logic/buildCache/AzureStorageBuildCacheProvider',
+  require
+);
+
+const PLUGIN_NAME: string = 'AzureStorageBuildCachePlugin';
+
+interface IAzureStorageConfigurationJson {
+  /**
+   * The name of the the Azure storage account to use for build cache.
+   */
+  storageAccountName: string;
+
+  /**
+   * The name of the container in the Azure storage account to use for build cache.
+   */
+  storageContainerName: string;
+
+  /**
+   * The Azure environment the storage account exists in. Defaults to AzureCloud.
+   */
+  azureEnvironment?: AzureEnvironmentNames;
+
+  /**
+   * An optional prefix for cache item blob names.
+   */
+  blobPrefix?: string;
+
+  /**
+   * If set to true, allow writing to the cache. Defaults to false.
+   */
+  isCacheWriteAllowed?: boolean;
+}
+
+class AzureStorageBuildCachePlugin implements IRushPlugin {
+  public pluginName: string = PLUGIN_NAME;
+
+  private static _jsonSchema: JsonSchema = JsonSchema.fromFile(
+    path.join(__dirname, 'schemas', 'azure-blob-storage-config.schema.json')
+  );
+
+  public apply(rushSession: RushSession, rushConfig: RushConfiguration): void {
```

Comment:
I noticed the method name `apply` used several times in this PR; I'd vote for something more like `initialize` (`apply`, like `bind` and `call`, has such a specific meaning in JavaScript).

Reply:
Originally, `apply` comes from `webpack`(which built the `tapable` packages). Each wepback plugin has a `apply` function to enhance `webpack` behaviour. it feels like "plugin" apply on "compiler". IMO, it probably lower some barriers for those who are familiar with webpack plugin already. 

For instance, https://github.com/webpack/webpack/blob/main/lib/DelegatedPlugin.js

Anyway, I am open to rename this method to `initialize` if it would be confusing.

Reply:
My preference would be to leave it as `apply` to match webpak and Heft.